src/transaction/declare_v2.rs

```diff
@@ -42,15 +43,61 @@ pub struct DeclareV2 {
     pub signature: Vec<Felt252>,
     pub nonce: Felt252,
     pub compiled_class_hash: Felt252,
-    pub sierra_contract_class: SierraContractClass,
+    pub contract_class: MaybeSierraContractClass,
     pub sierra_class_hash: Felt252,
     pub hash_value: Felt252,
-    pub casm_class: Option<CasmContractClass>,
     pub skip_validate: bool,
     pub skip_execute: bool,
     pub skip_fee_transfer: bool,
 }
 
+#[derive(Debug, Clone)]
+pub enum MaybeSierraContractClass {
+    Sierra(SierraContractClass),
+    SierraAndCasm(SierraContractClass, CasmContractClass),
+    Casm(CasmContractClass),
+}
```

Comment:
These values are massive. Consider using `Arc` or `Box` for the variants to avoid huge copies when moving, e.g.:
```suggestion
pub enum MaybeSierraContractClass {
    Sierra(Arc<SierraContractClass>),
    SierraAndCasm(Arc<(SierraContractClass, CasmContractClass)>),
    Casm(Arc<CasmContractClass>),
}
```

Reply:
Fixed.